lib/peep.rb

```diff
@@ -0,0 +1,56 @@
+require 'pg'
+
+class Peep
+  attr_reader :id, :username, :message, :datetime
+
+  def initialize(id:, username:, message:, datetime:)
+    @id = id
+    @username = username
+    @message = message
+    @datetime = datetime
+  end
+
+  def self.all
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec('SELECT * FROM peeps ORDER BY datetime DESC;')
+    
+    result.map do |peep| Peep.new(id: peep['id'], username: peep['username'], 
+      message: peep['message'], datetime: peep['datetime'])
+    end
+  end
+
+  def self.add(username:, message:)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec_params(
+      "INSERT INTO peeps (username, message, datetime) VALUES ($1, $2, '#{Time.new}') 
+      RETURNING id, username, message;", [username, message])
+    
+    Peep.new(id: result[0]['id'], username: result[0]['username'], 
+      message: result[0]['message'], datetime: result[0]['datetime'])
+  end
+
+  def self.search(keyword)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec_params("SELECT * FROM peeps WHERE (message) 
```

Comment:
I think you could have just used the "exec" because you wouldn't be able to have a SQL injection here with the query as there's no insert.

Reply:
Give it a try :)

I think that the key is whether or not user input is incorporated into the query, since the intended query will be cut short by the SQL that a nefarious user injects.

So I think it's justified but... try it out! :)